op-supervisor/supervisor/backend/cross/safe_update_test.go

```diff
@@ -459,6 +459,9 @@ func TestScopedCrossSafeUpdate(t *testing.T) {
 		csd.openBlockFn = func(chainID eth.ChainID, blockNum uint64) (ref eth.BlockRef, logCount uint32, execMsgs map[uint32]*types.ExecutingMessage, err error) {
 			return opened, 0, execs, nil
 		}
+		csd.checkFn = func(chainID eth.ChainID, blockNum uint64, logIdx uint32, checksum types.MessageChecksum) (types.BlockSeal, error) {
+			return types.BlockSeal{}, errors.New("unexpected checkFn call. expected short-circuit")
+		}
```

Comment:
If this call is unexpected within the context of the test, returning an error may not have the desired effect, since it's the code-under-test that receives that error. Rather, I *think* you want to do some sort of `requires.Fail` (not sure the exact call) to make the *test fail* if this function were ever called.

Honestly it seems fine since the error check just below here correctly checks that it's a Conflict about "expired", so this is a non blocking nit.

Reply:
I like the explicitness using `require.Fail`. Fixed to use that. Thanks for the suggestion.